docs/getting_started/start_new_component.rst

```diff
@@ -0,0 +1,403 @@
+How to Start Working on a :ref:`Component <what_is_a_component>`
+################################################################
+
+To support :ref:`component <what_is_a_component>` development, SPL Core provides features to:
+
+- document the component behavior
+- test the component behavior
+- generate component report with full traceability between documentation, code and tests
+- **all** component artifacts are configurable (content will be included/excluded based on configuration)
+
+`Sphinx <https://www.sphinx-doc.org/>`_ is used to generate the component report.
+See the project `conf.py` file for the configuration details.
+
+Detailed Design
+***************
+
+The component behavior is described in restructured text in the ``doc`` directory.
+
+Here are the main features supporting component documentation:
+
+* describe the component behavior in text
+* create diagrams using `mermaid <https://mermaid-js.github.io/mermaid/>`_
+* include/exclude content based on configuration 
+* use `sphinx-needs <https://www.sphinx-needs.com/>`_ to create traceability between documentation, code and tests
+    
+
+**Traceability**
+
+In order to trace the documentation to the code or tests, you need to add a `sphinx-needs <https://www.sphinx-needs.com/>`_ element.
+
+.. code-block:: rst
+
+    .. spec:: Feature X
+        :id: SWDD_COMP-001
+
+        The component shall implement feature X.
+
+The ``spec`` is used to tag the component specification. It shall have:
+
+* a title to very briefly describe the specification. This will be shown in traceability tables.
+* an id to uniquely identify the specification. The id naming convention is ``SWDD_<COMPONENT_NAME>-<NUMBER>``

```

Comment:
Give hint, how one can determine the next "free ID".

Reply:
We should have a dedicated chapter where the component report feature is explained. There we can describe the traceability and how to keep the IDs unique and have a numbering scheme.